uploadNeuroDB/NeuroDB/objectBroker/MriUploadOB.pm

```diff
@@ -0,0 +1,180 @@
+package NeuroDB::objectBroker::MriUploadOB;
+
+=pod
+
+=head1 NAME
+
+NeuroDB::objectBroker::MriUploadOB -- An object broker for MRI uploads
+
+=head1 SYNOPSIS
+
+  use NeuroDB::Database;
+  use NeuroDB::objectBroker::MriUploadOB;
+  use TryCatch;
+
+  my $db = NeuroDB::Database->new(
+      userName     => 'user',
+      databaseName => 'my_db',
+      hostName     => 'my_hostname',
+      password     => 'pwd'
+  );
+
+  try {
+      $db->connect();
+  } catch(NeuroDB::DatabaseException $e) {
+      die sprintf(
+          "User %s failed to connect to %s on %s: %s (error code %d)\n",
+          'user',
+          'my_db',
+          'my_hostname',
+          $e->errorMessage,
+          $e->errorCode
+      );
+  }
+
+  .
+  .
+  .
+
+  my $mriUploadOB = NeuroDB::objectBroker::MriUploadOB->new(db => $db);
+  my $mriUploadsRef;
+  try {
+      $mriUploadsRef= $mriUploadOB->getWithTarchive(
+          1, '/tmp/my_tarchive.tar.gz', 1
+      );
+  } catch(NeuroDB::objectBroker::MriUploadOBException $e) {
+      die sprintf(
+          "Failed to retrieve MRI uploads: %s",
+          $e->errorMessage
+      );
+  }
+
+=head1 DESCRIPTION
+
+This class provides a set of methods to either fetch or insert mri upload
+records. The operations are always performed on database table C<mri_upload>.
+
+=head2 Methods
+
+=cut
+
+use Moose;
+use MooseX::Privacy;
+
+use NeuroDB::Database;
+use NeuroDB::DatabaseException;
+use NeuroDB::objectBroker::ObjectBrokerException;
```

Comment:
missing: `use File::Basename`
Otherwise, get the following error:
Undefined subroutine &NeuroDB::objectBroker::MriUploadOB::basename called at /data/preventAD/bin/mri/uploadNeuroDB/NeuroDB/objectBroker/MriUploadOB.pm line 121.
	...propagated at /data/preventAD/bin/mri/uploadNeuroDB/NeuroDB/objectBroker/MriUploadOB.pm line 133.
	...propagated at updateMRI_Upload.pl line 182.

Reply:
Huh? How come I didn't get this error on my end? Will add the "use..." as requested.

Reply:
mystery of computers ;)